knative-operator/pkg/controller/knativeserving/consoleclidownload/consoleclidownload.go

```diff
@@ -44,52 +40,55 @@ func Apply(instance *servingv1alpha1.KnativeServing, apiclient client.Client, sc
 	return nil
 }
 
-// reconcileKnCCDResources reconciles required resources viz Deployment, Service, Route
+// reconcileKnCCDResources reconciles required resources viz Knative Service
 // which will serve kn cross platform binaries within cluster
 func reconcileKnCCDResources(instance *servingv1alpha1.KnativeServing, apiclient client.Client, scheme *runtime.Scheme) error {
-	log.Info("Installing kn ConsoleCLIDownload resources")
-	manifest, err := manifest(instance, apiclient, scheme)
-	if err != nil {
+	if !instance.Status.IsReady() {
+		return fmt.Errorf("Knative instance %q/%q not ready yet", instance.Name, instance.GetNamespace())
```

Comment:
I don't think we need to return an error here since returning an error will retry immediately. Instead, we can just return `nil` and wait for the informer to kick us again once the KnativeServing became ready.

Reply:
Done in https://github.com/openshift-knative/serverless-operator/pull/381/commits/f79521812b43c1d2cad86c240249dbb3633dbb16.